lib/templateCompilerModules/utils.ts

```diff
@@ -8,16 +8,46 @@ export interface ASTNode {
   attrs: Attr[]
 }
 
+import {UrlWithStringQuery, parse as uriParse} from 'url';
+
 export function urlToRequire (url: string): string {
+  let returnValue = `"${url}"`
   // same logic as in transform-require.js
   const firstChar = url.charAt(0)
   if (firstChar === '.' || firstChar === '~' || firstChar === '@') {
     if (firstChar === '~') {
       const secondChar = url.charAt(1)
       url = url.slice(secondChar === '/' ? 2 : 1)
     }
-    return `require("${url}")`
-  } else {
-    return `"${url}"`
+
+    const uriParts = parseUriParts(url)
+
+    !uriParts.hash
+    ? returnValue = `require("${url}")`
+    : 
+    // support uri fragment case by excluding it from
+    // the require and instead appending it as string; 
+    // assuming that the path part is sufficient according to 
+    // the above caseing(t.i. no protocol-auth-host parts expected)
+    returnValue = `require("${uriParts.path}") + "${uriParts.hash}"`
   }
+  return returnValue
+}
+
+/**
+ * vuejs/component-compiler-utils#22 Support uri fragment in transformed require
+ * @param urlString an url as a string 
+ */
+function parseUriParts(urlString: string): UrlWithStringQuery{
+    // initialize return value
+    const returnValue : UrlWithStringQuery = uriParse("");
+    if(urlString){
```

Comment:
It seems that the code style is not consistent with the existing code. It seems that we should add a linting tool here. @znck

Reply:
replaced double quotes with single quotes, if that is all you have meant